source/iterator/cdc_iterator.go

```diff
@@ -0,0 +1,263 @@
+	iterator := &CDCIterator{
+		client:       client,
+		buffer:       make(chan Record, 1),
+		ticker:       time.NewTicker(pollingPeriod),
+		tomb:         &tomb.Tomb{},
+		fields:       fields,
+		lastModified: lastModifiedTime.Add(time.Second),
+	}
+	iterator.tomb.Go(func() error {
+		return iterator.poll(ctx)
+	})
+	return iterator, nil
+}
+
+func (c *CDCIterator) poll(ctx context.Context) error {
+	defer close(c.buffer)
+	for {
+		select {
+		case <-c.tomb.Dying():
+			return c.tomb.Err()
+		case <-c.ticker.C:
+			err := c.flushLatestLeads(ctx)
+			if err != nil {
+				return err
+			}
+		}
+	}
+}
+
+func (c *CDCIterator) HasNext(ctx context.Context) bool {
+	logger := sdk.Logger(ctx).With().Str("Method", "Has Next").Logger()
+	logger.Trace().Msg("Checking iterator has next record...")
+	return len(c.buffer) > 0 || !c.tomb.Alive() // if tomb is dead we return true so caller will fetch error with Next
+}
+
+func (c *CDCIterator) Next(ctx context.Context) (sdk.Record, error) {
+	select {
+	case r := <-c.buffer:
+		return c.prepareRecord(r)
+	case <-c.tomb.Dead():
+		return sdk.Record{}, c.tomb.Err()
+	case <-ctx.Done():
+		return sdk.Record{}, ctx.Err()
+	}
+}
+
+func (c *CDCIterator) Stop() {
+	// stop the two goRoutines
+	c.ticker.Stop()
+	c.tomb.Kill(errors.New("cdc iterator is stopped"))
+}
+
+func (c *CDCIterator) prepareRecord(r Record) (sdk.Record, error) {
+	key := strconv.Itoa(r.id)
+	if r.deleted {
+		position := position.Position{
+			Type:      position.TypeCDC,
+			Key:       key,
+			CreatedAt: time.Now().UTC(),
+			UpdatedAt: time.Now().UTC(),
+		}
+		pos, err := position.ToRecordPosition()
+		if err != nil {
+			return sdk.Record{}, err
+		}
+		return sdk.Record{
+			Key: sdk.StructuredData{
+				"id": key,
+			},
+			Position: pos,
+			Metadata: map[string]string{
+				"action": "delete",
+			},
+			CreatedAt: time.Now().UTC(),
+		}, nil
+	}
+	createdAt, err := time.Parse(time.RFC3339, fmt.Sprintf("%s", r.data["createdAt"]))
+	if err != nil {
+		return sdk.Record{}, fmt.Errorf("error parsing createdAt %w", err)
+	}
+	updatedAt, err := time.Parse(time.RFC3339, fmt.Sprintf("%s", r.data["updatedAt"]))
+	if err != nil {
+		return sdk.Record{}, fmt.Errorf("error parsing updatedAt %w", err)
+	}
+	position, _ := position.Position{
+		Type:      position.TypeCDC,
+		Key:       key,
+		CreatedAt: createdAt,
+		UpdatedAt: updatedAt,
+	}.ToRecordPosition()
+	rec := sdk.Record{
+		Payload: sdk.StructuredData(r.data),
+		Metadata: map[string]string{
+			"id":        key,
+			"createdAt": createdAt.UTC().Format(time.RFC3339),
+			"updatedAt": updatedAt.UTC().Format(time.RFC3339),
+		},
+		Position: position,
+		Key: sdk.StructuredData{
+			"id": key,
+		},
+	}
+
+	return rec, nil
+}
+
+func (c *CDCIterator) flushLatestLeads(ctx context.Context) error {
+	logger := sdk.Logger(ctx).With().Str("Method", "getLatestLeads").Logger()
+	logger.Trace().Msg("Starting the getLatestLeads")
```

Comment:
The method is actually  flushLatestLeads and not getLatestLeads.

Reply:
Updated

Reply:
Still the same: https://github.com/rustiever/conduit-connector-marketo/pull/5/files/9e6c5ddb96dfc09477c75fb8adcfc00ae28518ed#diff-e106e8cb0d1d35d3861681465c151caa8f9bf0e04f86db0831dee5dd0bf84bb9R154